agent/consul/state/config_entry.go

```diff
@@ -1435,6 +1441,28 @@ func readDiscoveryChainConfigEntriesTxn(
 		res.Services[svcID] = entry
 	}
 
+	peerEntMeta := structs.DefaultEnterpriseMetaInPartition(entMeta.PartitionOrDefault())
+	for peerName := range todoPeers {
+		q := Query{
+			Value:          strings.ToLower(peerName),
+			EnterpriseMeta: *peerEntMeta,
+		}
+		idx, entry, err := peeringReadTxn(tx, ws, q)
```

Comment:
We should do some basic validation to ensure that this is going to constantly wake up the watch-set and cause lots of recompilations.